pkg/parse/json/json_test.go

```diff
@@ -0,0 +1,552 @@
+	  "homepage": null,
+	  "size": 0,
+	  "stargazers_count": 0,
+	  "watchers_count": 0,
+	  "language": null,
+	  "has_issues": true,
+	  "has_projects": true,
+	  "has_downloads": true,
+	  "has_wiki": true,
+	  "has_pages": true,
+	  "forks_count": 0,
+	  "mirror_url": null,
+	  "archived": false,
+	  "disabled": false,
+	  "open_issues_count": 2,
+	  "license": null,
+	  "forks": 0,
+	  "open_issues": 2,
+	  "watchers": 0,
+	  "default_branch": "master"
+	},
+	"sender": {
+	  "login": "Codertocat",
+	  "id": 21031067,
+	  "node_id": "MDQ6VXNlcjIxMDMxMDY3",
+	  "avatar_url": "https://avatars1.githubusercontent.com/u/21031067?v=4",
+	  "gravatar_id": "",
+	  "url": "https://api.github.com/users/Codertocat",
+	  "html_url": "https://github.com/Codertocat",
+	  "followers_url": "https://api.github.com/users/Codertocat/followers",
+	  "following_url": "https://api.github.com/users/Codertocat/following{/other_user}",
+	  "gists_url": "https://api.github.com/users/Codertocat/gists{/gist_id}",
+	  "starred_url": "https://api.github.com/users/Codertocat/starred{/owner}{/repo}",
+	  "subscriptions_url": "https://api.github.com/users/Codertocat/subscriptions",
+	  "organizations_url": "https://api.github.com/users/Codertocat/orgs",
+	  "repos_url": "https://api.github.com/users/Codertocat/repos",
+	  "events_url": "https://api.github.com/users/Codertocat/events{/privacy}",
+	  "received_events_url": "https://api.github.com/users/Codertocat/received_events",
+	  "type": "User",
+	  "site_admin": false
+	}
+}`,
+	}
+	for i := 0; i < len(payloads); i++ {
+		byteMessage := []byte(payloads[i])
+		m := make(map[string]interface{})
+		err := gojson.Unmarshal(byteMessage, &m)
+		if err != nil {
+			t.Error("Failed to unmarshal payload")
+			continue
+		}
+		eventBindings, expectedValues := eventBindingDigger(t, m, parse.EventInterpolationBase)
+		t.Log("All Event Bindings")
+		for i := range eventBindings {
+			t.Log(eventBindings[i])
+		}
+		actualValues := FieldsFromByteMessage(byteMessage, eventBindings)
+		for i := range actualValues {
+			t.Logf("Validating %s", eventBindings[i])
+
+			// Remove spacing to normalize both values
+			actualBuffer := new(bytes.Buffer)
+			expectedBuffer := new(bytes.Buffer)
+			if err := gojson.Compact(actualBuffer, []byte(actualValues[i])); err != nil {
+				t.Errorf("Unable to compact actual bytes for field %s", eventBindings[i])
+				continue
+			}
+			if err := gojson.Compact(expectedBuffer, []byte(expectedValues[i])); err != nil {
+				t.Errorf("Unable to compact expected bytes for field %s", eventBindings[i])
+				continue
+			}
+			// Grab token to determine value type
+			expectedBufferCopy := bytes.NewBuffer(expectedBuffer.Bytes())
+			decoder := gojson.NewDecoder(expectedBufferCopy)
+			token, err := decoder.Token()
+			if err != nil {
+				t.Errorf("Error getting token from expected bytes for field %s", eventBindings[i])
+				continue
+			}
+			_, ok := token.(gojson.Delim)
+			if ok { // Compare json structures
```

Comment:
it's a bit confusing to me (probably b/c I have waaaay less context here :pray:) about what we're testing, e.g. what conditions would make this test fail? (probably could fix that with comments!)